network/controller/bootstrap/bootstrap.go

```diff
@@ -541,11 +553,39 @@ func (bc *bootstrapper) startBootstrap(ctx context.Context, address string) (*ne
 		// FirstPulseTime:    time.Unix(data.FirstPulseTimeUnix, 0),
 		Host:              response.GetSenderHost(),
 		ReconnectRequired: data.Code == ReconnectRequired,
+		NetworkSize:       data.NetworkSize,
 	}, nil
 }
 
+func (bc *bootstrapper) getNextBootstrapNodeIndex() int {
+	bc.nextBootstrapNodeIndex++
+	if bc.nextBootstrapNodeIndex >= len(bc.Certificate.GetDiscoveryNodes()) {
+		bc.nextBootstrapNodeIndex = 0
+	}
+	return bc.nextBootstrapNodeIndex
+}
+
+func (bc *bootstrapper) startCyclicBootstrap(ctx context.Context) {
+	for atomic.LoadInt32(&bc.cyclicBootstrapStop) == 0 {
```

Comment:
1. Get discovery nodes from certificate at the start of the method;
2. In every iteration try to bootstrap only to those N discovery nodes that are not active nodes yet;
3. Select node from N discovery nodes, whose NetworkSize is max. If its NetworkSize is larger than current node active node list size, then reconnect to that discovery node.

Reply:
done